app/allocations/middleware.test.js

```diff
@@ -0,0 +1,132 @@
+const middleware = require('./middleware')
+const allocationService = require('../../common/services/allocation')
+
+describe('#setAllocationsSummary', function() {
+  let next
+  let locals
+  beforeEach(async function() {
+    sinon.stub(allocationService, 'getCount').returns(18)
+    next = sinon.stub()
+    locals = {
+      dateRange: ['2020-01-01', '2020-01-10'],
+    }
+    await middleware.setAllocationsSummary(
+      {
+        params: {
+          date: '2020-01-01',
+        },
+        t: sinon.stub().returnsArg(0),
+      },
+      {
+        locals,
+      },
+      next
+    )
+  })
+  it('calls get count with the date range', function() {
+    expect(allocationService.getCount).to.have.been.calledOnceWithExactly({
+      dateRange: ['2020-01-01', '2020-01-10'],
+    })
+  })
+  it('updates the locals', function() {
+    expect(locals.allocationsSummary).to.deep.equal([
+      {
+        active: false,
+        label: 'allocations::dashboard.label',
+        href: '/allocations/week/2020-01-01/',
+        value: 18,
+      },
+    ])
+  })
+  it('calls next', function() {
+    expect(next).to.have.been.calledOnce
+  })
+})
+
+describe('#setPagination', function() {
+  const mockDateRange = ['2020-04-13', '2020-04-19']
+  let req, res, nextSpy
+  beforeEach(function() {
+    this.clock = sinon.useFakeTimers(new Date(mockDateRange[0]).getTime())
+    nextSpy = sinon.spy()
+  })
+
```

Comment:
Spotted a few times there's some inconsistency with new line spacing. I find it easier to scan when there is a line between blocks on the same level.

Not a massive deal but might look to see if there's some eslint rules to make it easier when swapping between files.

Reply:
I will add the line. We might add https://eslint.org/docs/rules/lines-between-class-members in our eslint config.